xbmc/windowing/win10/WinSystemWin10.cpp

```diff
@@ -692,7 +692,14 @@ float CWinSystemWin10::GetGuiSdrPeakLuminance() const
 */
 bool CWinSystemWin10::HasSystemSdrPeakLuminance()
 {
-  return CWIN32Util::GetSystemSdrWhiteLevel(std::wstring(), nullptr);
+  if (m_threadId == GetCurrentThreadId())
```

Comment:
Since the reason for this PR is multithreading, should catch cases where the window hasn't been initialized yet (just in case) to assert or log.

It doesn't look like any assumption can be made about the values of m_threadId, nothing in the doc says it cannot be 0
That would means a separate bool for validity or a more modern std::optional<DWORD>

Same comment for IsHDRDisplay().

Reply:
> the API cannot return errors and 0 is not a special value for "invalid".
> That would means a separate bool for validity or a more modern std::optional

This is not necessary because threadId never can be 0:

> The [CreateThread](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createthread) and [CreateRemoteThread](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createremotethread) functions also return an identifier that uniquely identifies the thread throughout the system. A thread can use the [GetCurrentThreadId](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentthreadid) function to get its own thread identifier. The identifiers are valid from the time the thread is created until the thread has been terminated. **Note that no thread identifier will ever be 0.**

https://learn.microsoft.com/en-us/windows/win32/procthread/thread-handles-and-identifiers

Reply:
OK I stand corrected. Is there a THREAD_INVALID or other define that would be a little better than hardcoding 0?

Reply:
Not that I know